lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2330,6 +2335,88 @@ decodeTransaction ctx (ApiT wid) (ApiSerialisedTransaction (ApiT sealed)) = do
             DelegationCertificate $
             JoinPoolExternal (ApiT rewardKey, Proxy @n) (ApiT poolId')
 
+submitTransaction
+    :: forall ctx s k (n :: NetworkDiscriminant).
+        ( ctx ~ ApiLayer s k
+        , HasNetworkLayer IO ctx
+        , IsOwned s k
+        , Typeable s
+        , Typeable n
+        )
+    => ctx
+    -> ApiT WalletId
+    -> ApiSerialisedTransaction
+    -> Handler ApiTxId
+submitTransaction ctx apiw@(ApiT wid) apitx@(ApiSerialisedTransaction (ApiT sealedTx)) = do
+    ttl <- liftIO $ W.getTxExpiry ti Nothing
+    apiDecoded <- decodeTransaction @_ @s @k @n ctx apiw apitx
+    when (isForeign apiDecoded) $
+        liftHandler $ throwE ErrSubmitTransactionForeignWallet
+    let ourOuts = getOurOuts apiDecoded
+    let ourInps = getOurInps apiDecoded
+
+    let allInpsNum = length $ apiDecoded ^. #inputs
+    let witsNum = length $ getSealedTxWitnesses sealedTx
+    when (allInpsNum > witsNum) $
+        liftHandler $ throwE $
+        ErrSubmitTransactionPartiallySignedOrNoSignedTx allInpsNum witsNum
```

Comment:
As we talked yesterday, at this moment we will recommend using submit-without-wid endpoint for anything outside integration tests (so "normal" txs at this moment). We will address all PAB / multisig cases as they appear. Right now all PAB integration tests pass with new endpoint. Here, we are just trying to say that when we have a specified number of "normal" inputs then sending tx that is partially signed or not signed at all does not make sense. I think it is reasonable FIRST approximation. The next moves will be:(a) deal with collaterals, (b) deal with script inputs (here sometimes as you say several wits will be required -> we will need to have a way to decode that first then enhance condition). Please notice that we are proceeded step by step and until we have not whole new tx workflow finished old endpoints and proxy submit is recommended to use.

Reply:
>As we talked yesterday, at this moment we will recommend using submit-without-wid endpoint for anything outside integration tests (so "normal" txs at this moment). 

My understanding was that we rather want to recommend submit-with-wid moving forward, especially in the construct-sign-submit flow? However if this particular issue is too big to address here, then I'd suggest capturing it in the Jira ticket? I understand that it would only affect balance-sign-submit flow?